gavel_ci/templates/console.html

```diff
@@ -0,0 +1,83 @@
+{% extends "layout.html" %}
+{% block body %}
+<h1>Gavel CI Build: <a href="{{url_for('jobserv.project', proj=proj)}}">{{proj}}</a> <a href="{{url_for('jobserv.build',proj=proj, build=build)}}">#{{build}}</a> - {{run}}</h1>
+<h3>
+    console
+    <span id='console-raw' style="display:none;">
+        (<a 
+          href="{{url_for('jobserv.run_artifact', proj=proj, build=build, run=run, p='console.log')}}"
+          download="{{proj}}-{{build}}-{{run}}-console.log">raw</a>)
+    </span>
+    <span id='console-status'></span>
+</h3>
+<div>
+    <pre id="console-log-tail"></pre>
+</div>
+{% endblock %}
+{% block script %}
+{{super()}}
+<script type="text/javascript">
+(function () {
+    let tailElement = $("#console-log-tail");
+    let downloadElement = $("#console-raw");
+    let statusElement = $("#console-status");
+    var offset = 0;
+    let messageTailWindow = 40;
+    var messageTail = [];
+    var status = '';
+    var interval = 3000; /* 3s */
+    let url = "/projects/{{proj}}/builds/{{build}}/{{run}}/console/tail";
+    var showStatus = function (slug) {
+        let msg = slug ? `: ${slug}` : '';
+        $(statusElement).text(msg);
+    };
+    var showDownload = function () {
+        $(downloadElement).show();
+    }
+    var updateLog = function (message) {
+        if (!message) return;
+        const messageRows = message.split('\n');
+        messageTail = messageTail.concat(messageRows);
+        messageTail = messageTail.slice(Math.max(messageTail.length - messageTailWindow, 0));
+        let messageContent = messageTail.join('\n');
+        console.log(`PANTERA displaying ${messageTail.length}`);
```

Comment:
looks to be some debug you can remove. Good band choice. The multimedia tech-lead back in the old Linaro days became a friend of mine. He'd played in a metal band around the metroplex and had been friends with dimebag.

Reply:
lol - totally forgot to remove this comment :D